homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -663,17 +769,45 @@ class PipelineInput:
     async def execute(self) -> None:
         """Run pipeline."""
         self.run.start()
-        current_stage = self.run.start_stage
+        current_stage: PipelineStage | None = self.run.start_stage
+        audio_buffer: list[bytes] = []
 
         try:
+            if current_stage == PipelineStage.WAKE:
+                assert self.stt_stream is not None
+                detect_result = await self.run.wake_word_detection(
```

Comment:
We should allow the caller to pass in a timeout so we stop detecting a wake word if nothing comes in for X seconds.

Reply:
Added a `timeout` to the wake word stage input